pkgs/shells/nushell/default.nix

```diff
@@ -36,16 +35,14 @@ rustPlatform.buildRustPackage rec {
 
   cargoBuildFlags = lib.optional withStableFeatures "--features stable";
 
-  cargoTestFlags = lib.optional withTestBinaries "--features test-bins";
-
   preCheck = ''
     export HOME=$TMPDIR
   '';
 
   checkPhase = ''
     runHook preCheck
-    echo "Running cargo cargo test ${lib.strings.concatStringsSep " " cargoTestFlags} -- ''${checkFlags} ''${checkFlagsArray+''${checkFlagsArray[@]}}"
-    cargo test ${lib.strings.concatStringsSep " " cargoTestFlags} -- ''${checkFlags} ''${checkFlagsArray+"''${checkFlagsArray[@]}"}
+    echo "Running cargo cargo test -- ''${checkFlags} ''${checkFlagsArray+''${checkFlagsArray[@]}}"
+    cargo test -- ''${checkFlags} ''${checkFlagsArray+"''${checkFlagsArray[@]}"}
     runHook postCheck
   '';
```

Comment:
I think you can remove the remove the `checkPhase`. Besides that, lgtm

Reply:
The complete `checkPhase`?

Shouldn't the tests still be executed (including the now built-in testbinaries)?

Reply:
>the complete checkPhase?

yes, `buildRustPackage` by default sets the same checkPhase https://github.com/NixOS/nixpkgs/blob/9f648fbc95f4931d3299b73ecc02b2ddb3d74bab/pkgs/build-support/rust/default.nix#L191-L196
IIRC nushell used a custom checkPhase because `buildRustPackage` doesn't support passing arguments directly to "cargo test"

Reply:
Ah thanks for the info, that's cool!

Amended this change and rebased to master